src/app/components/TradingChart/helpers.ts

```diff
@@ -0,0 +1,273 @@
+import { ApolloClient, gql } from '@apollo/client';
+import {
+  CandleDetails,
+  CandleDuration,
+} from 'app/pages/PerpetualPage/hooks/graphql/useGetCandles';
+import { Asset } from 'types';
+import { getTokenContract } from 'utils/blockchain/contract-helpers';
+import { TradingCandleDictionary } from './dictionary';
+
+export const MAX_CANDLE_COUNT = 500;
+export const graph_endpoint: string = process.env.REACT_APP_GRAPH_RSK!;
+
+export type Bar = {
+  time: number;
+  low: number;
+  high: number;
+  open: number;
+  close: number;
+  volume?: number;
+};
+
+export type CandleSticksResponse = {
+  id: string;
+  open: number;
+  high: number;
+  low: number;
+  close: number;
+  totalVolume: number;
+  periodStartUnix: number;
+};
+
+export const hasDirectFeed = (symbolName: string) => {
+  const [, quote] = symbolName.split('/') as [Asset, Asset];
+  return [Asset.RBTC, Asset.XUSD].includes(quote);
+};
+
+const queryDirectPair = async (
+  client: ApolloClient<any>,
+  candleDetails: CandleDetails,
+  baseToken: string,
+  quoteToken: string,
+  startTime: number,
+  endTime: number,
+  candleNumber: number,
+): Promise<CandleSticksResponse[]> => {
+  const query = gql`
+      {
+        candleSticks (
+          where: {
+            baseToken: "${baseToken}"
+            quoteToken: "${quoteToken}"
+            interval: "${candleDetails.entityName}"
+            periodStartUnix_gte: ${startTime}
+            periodStartUnix_lte: ${endTime}
+          }
+          orderBy: periodStartUnix
+          orderDirection: desc
+          first: ${Math.min(candleNumber, MAX_CANDLE_COUNT)}
+        ) {
+          id
+          open
+          high
+          low
+          close
+          totalVolume
+          periodStartUnix
+        }
+      }
+    `;
+  return client
+    .query({
+      query,
+    })
+    .then(response => response.data?.candleSticks || []);
```

Comment:
```suggestion
    .then(
      response =>
        response.data?.candleSticks?.map(item => ({
          ...item,
          open: Number(item.open),
          high: Number(item.high),
          low: Number(item.low),
          close: Number(item.close),
          totalVolume: Number(item.totalVolume),
        })) || [],
    )
```

Currently volume does not show up on the chart even when though data is returned from Graph query in `totalVolume` prop. This is due to the `volume` prop passed to the TradingView chart in candle data needing to be type `number` rather than a string. Simple fix is to convert the volume value in `queryCandles()`:
```volume: Number(item.totalVolume.toString()),``` or use suggested code change here.

However, this isn't the ideal solution as the `CandleSticksResponse` type does not have correct types for this prop, or `open`, `low,` `high`, and `close`. These values are never parsed into number format, as the data returned from Graph query is meant to already be in number format as per defined type. Ideally we would have these values returned as numbers directly by Graph, but if this is not possible then we should update typings/logic on FE (e.g. in `fillItem()`).